src/Raven.Server/Documents/Handlers/Batches/Commands/MergedBatchCommand.cs

```diff
@@ -155,15 +156,19 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                     break;
 
                 case CommandType.AttachmentPUT:
-                    attachmentIterator.MoveNext();
-                    var attachmentStream = attachmentIterator.Current;
-                    var stream = attachmentStream.Stream;
-                    _toDispose.Add(stream);
-
                     var docId = EtlGetDocIdFromPrefixIfNeeded(cmd.Id, cmd, lastPutResult);
 
-                    var attachmentPutResult = Database.DocumentsStorage.AttachmentsStorage.PutAttachment(context, docId, cmd.Name,
-                        cmd.ContentType, attachmentStream.Hash, cmd.ChangeVector, stream, updateDocument: false, extractCollectionName: ModifiedCollections is not null);
+                    Stream stream = null;
+
+                    if (cmd.RetireParameters.IsLocalAttachment())
```

Comment:
Under what conditions can we have a `PUT` for a retired attachment?
When we have an ETL from another RavenDB?
Can you add a comment explaining this?

Also `IsLocalAttachment` is a strange name for something that is being _uploaded_ here.

Reply:
added comments, changed the name from `IsLocalAttachment` to `IsLocalStorageAttachment`